core/adapters/bridge.go

```diff
@@ -31,59 +31,81 @@ func (ba *Bridge) Perform(input models.RunResult, store *store.Store) models.Run
 	} else if input.Status.PendingBridge() {
 		return resumeBridge(input)
 	}
-	ba.handleNewRun(&input, store.Config.BridgeResponseURL())
-	return input
+	return ba.handleNewRun(input, store.Config.BridgeResponseURL())
 }
 
 func resumeBridge(input models.RunResult) models.RunResult {
 	input.Status = models.RunStatusInProgress
 	return input
 }
 
-func (ba *Bridge) handleNewRun(input *models.RunResult, bridgeResponseURL *url.URL) {
+func (ba *Bridge) handleNewRun(input models.RunResult, bridgeResponseURL *url.URL) models.RunResult {
 	if ba.Params == nil {
 		ba.Params = new(models.JSON)
 	}
 	var err error
 	if input.Data, err = input.Data.Merge(*ba.Params); err != nil {
-		input.SetError(baRunResultError("handling data param", err))
-		return
+		var output models.RunResult
+		output.Data = input.Data
```

Comment:
This looks a little weird, in the case of an error we would save the data from the input to the output, I feel like in the case of an error we can leave Data empty? It should save space and is not needed.